src/contract_wrappers/modules/checkpoint/voting/advanced_p_l_c_r_voting_checkpoint_wrapper/common.ts

```diff
@@ -0,0 +1,1206 @@
+      keccakKeys.push('uint256');
+      return valueToWei(new BigNumber(e), FULL_DECIMALS).toString();
+    });
+
+    const secretVote = ethersUtils.solidityKeccak256(keccakKeys, [...bgVotes, params.salt]);
+
+    return (await this.contract).commitVote.sendTransactionAsync(
+      numberToBigNumber(params.ballotId),
+      secretVote,
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Used to reveal the vote
+   */
+  public revealVote = async (params: RevealVoteParams): Promise<PolyResponse> => {
+    await this.checkIndexOutOfBound(params.ballotId);
+    await this.checkValidStage(
+      {
+        ballotId: params.ballotId,
+      },
+      BallotStage.Reveal,
+    );
+
+    const ballot = await this.getBallotDetails({ ballotId: params.ballotId });
+    assert.assert(!ballot.isCancelled, ErrorCode.PreconditionRequired, 'Cancelled ballot');
+
+    const caller = await this.getCallerAddress(params.txData);
+    const commitVote: PendingBallots = await this.pendingBallots(caller);
+    const alreadyVoted = commitVote.pendingRevealBallots.find(v => {
+      return v === params.ballotId;
+    });
+    assert.assert(alreadyVoted !== undefined, ErrorCode.PreconditionRequired, 'Secret vote not available');
+
+    let choiceCount = 0;
+    for (let i = 0; i <= ballot.totalProposals - 1; i += 1) {
+      const noOfChoice = ballot.proposalChoicesCounts[i];
+      choiceCount += noOfChoice === 0 ? 3 : noOfChoice;
+    }
+    assert.assert(choiceCount === params.choices.length, ErrorCode.MismatchedArrayLength, 'Choices count mismatch');
+    const getVoteCommitEvents = await this.getLogsAsync({
+      eventName: AdvancedPLCRVotingCheckpointEvents_3_1_0.VoteCommit,
+      indexFilterValues: { _voter: caller },
+    });
+
+    const args = map(getVoteCommitEvents, e => {
+      return e.args;
+    });
+
+    const onlyBallot = find(args, e => {
+      /* eslint-disable no-underscore-dangle */
+      return e._ballotId.toNumber() === params.ballotId;
+    });
+    assert.assert(onlyBallot !== undefined, ErrorCode.NotFound, 'Ballot id not found');
+
+    const bgVotes = params.choices.map(e => valueToWei(new BigNumber(e), FULL_DECIMALS).toString());
+    const keccakKeys = times(bgVotes.length + 1, constant('uint256'));
+    const hash = ethersUtils.solidityKeccak256(keccakKeys, [...bgVotes, params.salt]);
+
+    assert.assert(
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      onlyBallot!._secretHash === hash,
+      ErrorCode.InvalidData,
+      'Invalid vote',
+    );
+
+    this.unsubscribeAll();
+
+    return (await this.contract).revealVote.sendTransactionAsync(
+      numberToBigNumber(params.ballotId),
+      params.choices.map(v => new BigNumber(valueToWei(new BigNumber(v), FULL_DECIMALS).toString())),
+      numberToBigNumber(params.salt),
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Allows the token issuer to scrapped down a ballot
```

Comment:
"scrapped down a ballot" I dont understand, can't we just say "cancel a ballot" ?